Core/detection.js

```diff
@@ -86,8 +86,12 @@
             duckduckgoMessaging.log(body)
             if (duckduckgoContentBlocking.shouldBlock(this.trackerUrl, "xmlhttprequest", function(url, block) { } )) { 
                 duckduckgoMessaging.log("blocking xhr")
-                xhr.abort()
-                return 
+                try { 
+                    xhr.abort() 
```

Comment:
I think this errors every time so I'm not sure that we are blocking XHR requests. Also, let's only send the success method back to the app when we know we have blocked something - in the error scenario we haven't blocked but we tell the app we have.